Tuist/ProjectDescriptionHelpers/AppCustomConfiguration.swift

```diff
@@ -0,0 +1,107 @@
+import ProjectDescription
+
+public enum AppCustomConfiguration {
+    case debug, betaDevelopment, betaStage, betaProduction, release
+    
+    internal func customConfiguration(with name: String, projectVersion: Version) -> CustomConfiguration {
+        switch self {
+        case .debug:
+            return CustomConfiguration.debug(name: configurationName, settings: settings(with: name, projectVersion: projectVersion))
+        case .betaDevelopment, .betaProduction, .betaStage, .release:
+            return CustomConfiguration.release(name: configurationName, settings: settings(with: name, projectVersion: projectVersion))
+        }
+    }
+    
+    internal func customTargetConfiguration(with name: String) -> CustomConfiguration {
+        switch self {
+        case .debug:
+            return CustomConfiguration.debug(name: configurationName, settings: targetSettings(with: name))
+        case .betaDevelopment, .betaProduction, .betaStage, .release:
+            return CustomConfiguration.release(name: configurationName, settings: targetSettings(with: name))
+        }
+    }
+    
+    private var configurationName: String {
+        switch self {
+        case .debug:
+            return "Debug"
+        case .betaDevelopment:
+            return "Beta-Development"
+        case .betaStage:
+            return "Beta-Stage"
+        case .betaProduction:
+            return "Beta-Production"
+        case .release:
+            return "Release"
+        }
+    }
+    
+    private func settings(with name: String, projectVersion: Version) -> [String: SettingValue] {
+        let base: [String: SettingValue] = [
+            "ACK_ENVIRONMENT_DIR": "$(PROJECT_DIR)/$(TARGET_NAME)/Environment",
+            "ACK_PROJECT_VERSION": SettingValue(stringLiteral: projectVersion.description),
+            "ACK_PRODUCT_NAME": SettingValue(stringLiteral: name),
```

Comment:
What is this for?

Reply:
This was due to the generalization I needed for a different project, but the way it is setup now it's not necessary, deleted, good catch 🙂